services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusDataTest.kt

```diff
@@ -24,293 +32,790 @@ import org.junit.jupiter.params.provider.NullSource
+                    )
+                private val base: KaivuilmoitusData =
+                    this@Kaivuilmoitus.base.copy(
+                        customerWithContacts =
+                            HakemusyhteystietoFactory.create(
+                                yhteyshenkilot = listOf(yhteyshenkilo1, yhteyshenkilo2)
+                            )
+                    )
+
+                private val withOneHenkilo =
                     base.copy(
-                        areas =
-                            listOf(
-                                ApplicationFactory.createCableReportApplicationArea(name = "first"),
-                                ApplicationFactory.createCableReportApplicationArea(
-                                    name = "removed1"
-                                ),
-                                ApplicationFactory.createCableReportApplicationArea(
-                                    name = "removed2"
-                                ),
-                                ApplicationFactory.createCableReportApplicationArea(name = "second"),
+                        customerWithContacts =
+                            base.customerWithContacts!!.copy(
+                                yhteyshenkilot = listOf(yhteyshenkilo1)
                             )
                     )
 
-                val result = base.listChanges(dataWithTwoAreas)
+                @Test
+                fun `returns a change when yhteyshenkilo removed from yhteystieto`() {
+                    val result = base.listChanges(withOneHenkilo)
+
+                    assertThat(result).containsExactly("customerWithContacts")
+                }
+
+                @Test
+                fun `returns a change when yhteyshenkilo added to yhteystieto`() {
+                    val result = withOneHenkilo.listChanges(base)
+
+                    assertThat(result).containsExactly("customerWithContacts")
+                }
+
+                @Test
+                fun `returns a change when yhteyshenkilo changed in yhteystieto`() {
+                    val updatedYhteyshenkilo = yhteyshenkilo1.copy(etunimi = "Muutettu")
+                    val updated =
+                        base.copy(
+                            customerWithContacts =
+                                base.customerWithContacts!!.copy(
+                                    yhteyshenkilot = listOf(yhteyshenkilo1, updatedYhteyshenkilo)
+                                )
+                        )
+
+                    val result = base.listChanges(updated)
 
-                assertThat(result).containsExactly("areas[1]", "areas[2]", "areas[3]")
+                    assertThat(result).containsExactly("customerWithContacts")
+                }
             }
 
-            @Test
-            fun `returns several changes when an area is added to the middle`() {
-                val updated =
+            @Nested
+            inner class AreaChanges {
```

Comment:
Since the code checking the area changes are in a different class now, maybe the test code should be for a different class as well?

Reply:
Created `HakemusalueTest` and placed tests that test a single `Hakemusalue` there. Kept those tests that test the area collection (changed the inner test class name to `AreasChanges`).